src/components/Main/Results/DeterministicLinePlot.tsx

```diff
@@ -97,6 +97,24 @@ function legendFormatter(enabledPlots: string[], value: string, entry: any) {
   return <span className={activeClassName}>{value}</span>
 }
 
+
+
+export class DebouncedDeterministicLinePlot extends React.Component {
+  constructor(props) {
+    super(props);
+    this.debouncedRender = _.debounce( this.forceUpdate, 1000)
+  }
+
+  shouldComponentUpdate() {
+    this.debouncedRender()
+    return false;
+  }
+
+  render() {
+    return <DeterministicLinePlot {...this.props}/>
+  }
+}
+
```

Comment:
The goal was a functional solution, but `DeterministicLinePlot` has a lot of state to wrangle. This class intercepts props changes coming from outside the component and debounces the update to 1000msec intervals.